src/components/ha-expansion-panel.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import "./ha-svg-icon";
+import { mdiChevronDown } from "@mdi/js";
+import { classMap } from "lit-html/directives/class-map";
+
+@customElement("ha-expansion-panel")
+class HaExpansionPanel extends LitElement {
+  @property({ type: Boolean, reflect: true }) expanded = false;
+
+  @property({ type: Boolean, reflect: true }) outlined = false;
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="summary" @click=${this._toggleContainer}>
+        <div class="summary-content">
+          <slot name="title"></slot>
+        </div>
+        <div class="summary-icon ${classMap({ expanded: this.expanded })}">
+          <ha-svg-icon .path=${mdiChevronDown}></ha-svg-icon>
+        </div>
+      </div>
+      <div class="container ${classMap({ expanded: this.expanded })}">
+        <slot name="content"></slot>
+      </div>
+    `;
+  }
+
+  private _toggleContainer(): void {
+    this.expanded = !this.expanded;
+    fireEvent(this, this.expanded ? "expanded" : "collapsed");
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        display: block;
+      }
+
+      :host([outlined]) {
+        box-shadow: none;
+        border-width: 1px;
+        border-style: solid;
+        border-color: var(
+          --ha-card-border-color,
+          var(--divider-color, #e0e0e0)
+        );
+      }
+
+      .summary {
+        display: flex;
+        padding: 0px 16px;
+        min-height: 48px;
+        position: relative;
+        align-items: center;
+        cursor: pointer;
+      }
+
+      .summary-content {
+        margin: 12px 0;
+        display: flex;
+        flex-grow: 1;
+        overflow: hidden;
+      }
+
+      ::slotted([slot="title"]) {
+        display: flex;
+      }
+
+      .summary-icon {
+        transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1);
+      }
+
+      .summary-icon.expanded {
+        transform: rotate(180deg);
+      }
+
+      .container {
+        overflow: hidden;
+        transition: max-height 200ms cubic-bezier(0.4, 0, 0.2, 1);
+        max-height: 0px;
+      }
+
+      .container.expanded {
+        max-height: 1000px;
```

Comment:
This should not be a max-height of 1000px, what if I put 2000px of content in it?

Reply:
I've tried animating height from 0 to auto, but that doesn't work :/
The current solution is using max-height. Another option is to use JS to animate the height, but I won't know the height of content if it will be lazy-loaded.
I'm open for suggestions on this.

Reply:
Use `display: none; overflow: hidden;` and `display: block;` instead of the height.

Reply:
This doesn't work :/ I want the slide up/down animation.

Reply:
You should change your transition, it now only works for max-height

Reply:
Setting it to 1000 will also not animate it correctly right? As it will animate it to 1000 and it will not be height 1000?

Reply:
Best would probably be to use the height, and remove the height on `transitionend`

Reply:
> Setting it to 1000 will also not animate it correctly right? As it will animate it to 1000 and it will not be height 1000?

RIght, the animation looks different if I use 150 and 5000 as max-height :/

> Best would probably be to use the height, and remove the height on transitionend

But I need to know the height of the content before I start the animation.
I'll search for something over the net

Reply:
Use `scrollHeight`

Reply:
ok, I got something. Maybe a bit overcomplicated, but it looks and works fine.
I'm not sure how to set the height of an already expanded panel. I used `firstUpdated`. Had to use `setTimeout` because without it height was not accurate (for example, scrollHeight returned 70 instead of 120)

Reply:
this is fixed with `transitionend` event - I'm setting `height: auto`